plugins/inputs/ctrlx_datalayer/ctrlx_datalayer_subscription.go

```diff
@@ -0,0 +1,188 @@
+package ctrlx_datalayer
+
+import (
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+// A subscription can be used to watch multiple ctrlX Data Layer nodes for changes.
+// Additional configuration settings can be given to tune the sampling and monitoring behaviour of the nodes.
+// All nodes in a subscription share the same configuration.
+// The plugin is able to create and manage multiple subscriptions.
+
+// The allowed values of the subscription property 'QueueBehaviour'
+var queueBehaviours = []string{"", "DiscardOldest", "DiscardNewest"}
+
+// The allowed values of the subscription property 'ValueChange'
+var valueChanges = []string{"", "Status", "StatusValue", "StatusValueTimestamp"}
```

Comment:
These are checked after default value is applied, so empty string can be removed from the list.
```suggestion
// The allowed values of the subscription property 'QueueBehaviour'
var queueBehaviours = []string{"DiscardOldest", "DiscardNewest"}

// The allowed values of the subscription property 'ValueChange'
var valueChanges = []string{"Status", "StatusValue", "StatusValueTimestamp"}
```

Reply:
Done, see https://github.com/influxdata/telegraf/pull/11155/commits/840b36589ab52f80bb125e47108d403c0a97414f